app/src/main/java/ch/epfl/sweng/rps/services/OfflineGameService.kt

```diff
@@ -0,0 +1,105 @@
+package ch.epfl.sweng.rps.services
+
+import ch.epfl.sweng.rps.models.ComputerPlayer
+import ch.epfl.sweng.rps.models.Game
+import ch.epfl.sweng.rps.models.Hand
+import ch.epfl.sweng.rps.models.Round
+import com.google.firebase.Timestamp
+import com.google.firebase.firestore.FirebaseFirestoreException
+import java.util.*
+
+class OfflineGameService(
+    private val gameId_: String,
+    private val uid: String,
+    private val computerPlayers: List<ComputerPlayer>,
+    private var roundCount: Int
+) : GameService {
```

Comment:
I don't think these should be the arguments of the class. I would say only the game mode makes sense as the argument. From this you can derive everything you need, generate a game id, etc.

This logic would be most likely be done in `startListening()`.

Reply:
Am I supposed to get the computer players there and userId?